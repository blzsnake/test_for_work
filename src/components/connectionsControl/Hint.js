import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Text } from 'ghofi-react-components';


import s from './styles.less';

/**
 * @param  {string} {title=''
 * @param  {array} texts=[]}
 */

const Hint = ({ texts = [] }) => (
  <div className={s.hint}>
    <div className={s.hint__title}>
      <Text
        size={36}
        sizeTablet={24}
        isThreeSixB >
        <FormattedMessage id='connections_control_hint_title' />
      </Text>
    </div>
    <div className={s.hint__texts}>
      {texts.map((text, index) => (
        <Text
          size={18}
          sizeTablet={16}
          tagName='p'>
          <FormattedMessage id={`connections_control_hint_text_${index + 1}`} />
        </Text>))}
    </div>
  </div>
);

export default Hint;
