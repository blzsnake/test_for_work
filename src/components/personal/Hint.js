import React from 'react';
import classnames from 'classnames';
import { FormattedMessage } from 'react-intl';

import { Text } from 'ghofi-react-components';

import s from './styles.less';

const makeInfo = texts => (
  <div className={s.hint__info}>{texts.map(text => (
    <div className={s.hint__paragraph}>
      <Text
        size={18}
        tagName='p'>
        <FormattedMessage id={text} />
      </Text>
    </div>))}
  </div>
);

const makeLegend = list => (
  <ul className={s.hint__legend}>{list.map(item => (
    <div className={s.hint__item}>
      <div className={classnames({
        [s.hint__view]: true,
        [s[`hint__view_${item.view}`]]: true,
        })} />
      <div className={s.hint__descr}>
        <Text
          size={18}
          tagName='p'>
          <FormattedMessage id={item.text} />
        </Text>
      </div>
    </div>))}
  </ul>
);

const Hint = ({ title = '', texts = [], list = [] }) => (
  <div className={s.hint}>
    <div className={s.hint__title}>
      <Text
        size={36}
        isThreeSixB >
        <FormattedMessage id={title} />
      </Text>
    </div>
    <div className={s.hint__wrapper}>
      {makeInfo(texts)}
      {makeLegend(list)}
    </div>
  </div>
);

export default Hint;
