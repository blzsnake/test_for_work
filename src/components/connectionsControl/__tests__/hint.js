import React from 'react';
import { shallow } from 'enzyme';

import Hint from '../Hint';

const props = {
  title: 'Mark Yuliy Cesar',
  texts: ['string'],
};

describe('Should render', () => {
  it('Hint render', () => {
    const component = shallow(<Hint {...props} />);

    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
