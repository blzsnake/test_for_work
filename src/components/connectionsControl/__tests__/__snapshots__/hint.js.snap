// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Hint render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hint
    texts={
      Array [
        "string",
      ]
    }
    title="Mark Yuliy Cesar"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
        >
          <Styled(Text)
            isThreeSixB={true}
            size={36}
            sizeTablet={24}
          >
            <FormattedMessage
              id="connections_control_hint_title"
              values={Object {}}
            />
          </Styled(Text)>
        </div>,
        <div
          className={undefined}
        >
          <Styled(Text)
            size={18}
            sizeTablet={16}
            tagName="p"
          >
            <FormattedMessage
              id="connections_control_hint_text_1"
              values={Object {}}
            />
          </Styled(Text)>
        </div>,
      ],
      "className": "hint",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Styled(Text)
            isThreeSixB={true}
            size={36}
            sizeTablet={24}
          >
            <FormattedMessage
              id="connections_control_hint_title"
              values={Object {}}
            />
          </Styled(Text)>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FormattedMessage
              id="connections_control_hint_title"
              values={Object {}}
            />,
            "isThreeSixB": true,
            "size": 36,
            "sizeTablet": 24,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "connections_control_hint_title",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Styled(Text)
              size={18}
              sizeTablet={16}
              tagName="p"
            >
              <FormattedMessage
                id="connections_control_hint_text_1"
                values={Object {}}
              />
            </Styled(Text)>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FormattedMessage
                id="connections_control_hint_text_1"
                values={Object {}}
              />,
              "size": 18,
              "sizeTablet": 16,
              "tagName": "p",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "connections_control_hint_text_1",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          >
            <Styled(Text)
              isThreeSixB={true}
              size={36}
              sizeTablet={24}
            >
              <FormattedMessage
                id="connections_control_hint_title"
                values={Object {}}
              />
            </Styled(Text)>
          </div>,
          <div
            className={undefined}
          >
            <Styled(Text)
              size={18}
              sizeTablet={16}
              tagName="p"
            >
              <FormattedMessage
                id="connections_control_hint_text_1"
                values={Object {}}
              />
            </Styled(Text)>
          </div>,
        ],
        "className": "hint",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Styled(Text)
              isThreeSixB={true}
              size={36}
              sizeTablet={24}
            >
              <FormattedMessage
                id="connections_control_hint_title"
                values={Object {}}
              />
            </Styled(Text)>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FormattedMessage
                id="connections_control_hint_title"
                values={Object {}}
              />,
              "isThreeSixB": true,
              "size": 36,
              "sizeTablet": 24,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "connections_control_hint_title",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Styled(Text)
                size={18}
                sizeTablet={16}
                tagName="p"
              >
                <FormattedMessage
                  id="connections_control_hint_text_1"
                  values={Object {}}
                />
              </Styled(Text)>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <FormattedMessage
                  id="connections_control_hint_text_1"
                  values={Object {}}
                />,
                "size": 18,
                "sizeTablet": 16,
                "tagName": "p",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "connections_control_hint_text_1",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
